lib/cmd/common/build.js

```diff
@@ -198,42 +198,55 @@ function mutateGitRefArgs(args) {
 function doBuild(args, cb) {
   var uri = "box/srv/1.1/wid/" + fhc.curTarget + "/" + args.destination + "/" + args.app + "/deliver";
   var doCall = function() {
-    var payload = argsToPayload(args);
-    common.doApiCall(fhreq.getFeedHenryUrl(), uri, payload, i18n._("Error reading app: "), function(err, data) {
-      if (err) {
-        return cb(err);
-      }
-      var keys = [];
-      if (data.cacheKey) keys.push(data.cacheKey);
-      //      if(data.stageKey) keys.push(data.stageKey);
-      if (keys.length > 0) {
-        async.map(keys, common.waitFor, function(err, results) {
+
+    //Fix missing creds names - RHMAP-9456
+    if (args.bundleId) {
+      common.findCredentialsByID(args.bundleId, function(err, data) {
+        if (!err) {
+          //If this is not set its just set as blank
+          if (data.bundleName) {
+            args.credentialName = data.bundleName;
+          }
+        }
+
+        var payload = argsToPayload(args);
+        common.doApiCall(fhreq.getFeedHenryUrl(), uri, payload, i18n._("Error reading app: "), function(err, data) {
           if (err) {
             return cb(err);
           }
-          if (results[0] && results[0][0] && results[0][0].action) {
-            var build_asset = results[0][0].action.url;
-            build.message = i18n._("\nDownload URL: ") + build_asset;
-            downloadBuild(args, build_asset, './', function(err, data) {
+          var keys = [];
+          if (data.cacheKey) keys.push(data.cacheKey);
+          //      if(data.stageKey) keys.push(data.stageKey);
+          if (keys.length > 0) {
+            async.map(keys, common.waitFor, function(err, results) {
               if (err) {
                 return cb(err);
               }
-              if (data) {
-                results.push({
-                  download: data
+              if (results[0] && results[0][0] && results[0][0].action) {
+                var build_asset = results[0][0].action.url;
+                build.message = i18n._("\nDownload URL: ") + build_asset;
+                downloadBuild(args, build_asset, './', function(err, data) {
+                  if (err) {
+                    return cb(err);
+                  }
+                  if (data) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)